examples/src/main/java/com/mapbox/navigation/examples/core/MapboxNavigationViewActivity.kt

```diff
@@ -11,6 +11,8 @@ class MapboxNavigationViewActivity : AppCompatActivity() {
         val binding = LayoutActivityNavigationViewBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
+        onBackPressedDispatcher.addCallback(this, binding.root.getOnBackPressedCallback())
```

Comment:
Few ideas to automate this.
1. Add `MapboxNavigation` extension method that will add `OnBackPressedCallback` to `AppCompatActivity.onBackPressedDispatcher`.
```kotlin
fun MapboxNavigation.initDropInUI(activity: AppCompatActivity, navView: DropInNavigationView) {
    activity.onBackPressedDispatcher.addCallback(activity, navView.getOnBackPressedCallback())
}
```
2. Add `AppCompatActivity` extension method that takes only `DropInNavigationView` arg.
```kotlin
fun AppCompatActivity.initDropInUI(navView: DropInNavigationView) {
    onBackPressedDispatcher.addCallback(this, navView.getOnBackPressedCallback())
}
```
3. Use Application.ActivityLifecycleCallbacks and AndroidManifest to `<meta-data>` tags to identify which activity is a NavigationView host and for which we should  register OnBackPressedCallback to intercept onBackPress. This solution would only work with activities that subclass `AppCompatActivity`.

Reply:
We can also capture the back stack inside the view. This could remove the need to add anything to the hosting activity/fragment

Prototyped an idea and [left it in documentation](https://docs.google.com/document/d/1u7T_tuTBK2iLHr6yhYQOSjb7ldVDy6rDuiz0kQCMn7I/edit?usp=sharing)
``` kotlin
@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
class NavigationBackStack(val view: View) : MapboxNavigationObserver {
    override fun onAttached(mapboxNavigation: MapboxNavigation) {
        view.isFocusableInTouchMode = true
        view.requestFocus()
        view.setOnKeyListener { _, keyCode, event ->
            if (event.action == KeyEvent.ACTION_UP && keyCode == KeyEvent.KEYCODE_BACK) {
                logger.e(TAG, Message("TODO handle back pressed"))
                false
            } else {
                false
            }
        }
    }

    override fun onDetached(mapboxNavigation: MapboxNavigation) {
        view.isFocusableInTouchMode = false
        view.setOnKeyListener { _, _, _ -> false }
    }

    private companion object {
        private val TAG = Tag("MbxNavigationBackStack")
    }
}
```

Reply:
Let do this in a separate PR.